### DISCUSSION TITLE
Anybody run the outline in local development successfully?

### DISCUSSION THREAD

**Discussion by Soyn** [outline/outline#2169] [Self Hosting] [answered]
- Follow the instruction, the slack redirect url can not be set as `http://*`, and after i fix the redirect URL issue.
- When do authenticated with slack. Error occurs!

### env
- OS: macOs Big Sur (Apple M1)
- codebase: `main`

### screenshoot
![image](https://user-images.githubusercontent.com/7734046/120665684-948c8700-c4be-11eb-9a3e-745fe488b44e.png)

**Comment by tommoor** [comment, marked as answer]
They `SECRET_KEY` must be a 32 character hexadecimal string, it's best to generate a compatible key with the command detailed in the instructions: `openssl rand -hex 32`

**Reply by Soyn** [reply]
Thanks. I have fixed this. I fill the value with random key directly, which occurs the error. And the local instruction is terrible. Cannot run in windows, and the Slack OAuth cannot be set by `http`. I will send a pull request to fix the Slack OAuth issue.

**Reply by tommoor** [reply]
I have improved the comment on the environment variable comment to better communicate the intended use and format:

https://github.com/outline/outline/commit/1f49b35c89ab0d85d7c78672b40dce501460e15c
